### PR TITLE
Deprecate PngEncoder::encode and IcoEncoder::encode

### PR DESCRIPTION
**Pull request by fintelia**
It seems to be the only function that expects input in big endian, so getting
rid of it will make the rest of the library more consistent.

### CODE REVIEW

File: src/codecs/png.rs [hunk 1]
```diff
@@ -575,6 +578,11 @@ impl<W: Write> PngEncoder<W> {
 }
 
 impl<W: Write> ImageEncoder for PngEncoder<W> {
+    /// Write a PNG image with the specified width, height, and color type.
+    ///
+    /// For color types with 16-bit per channel or larger, the contents of `buf` should be in
+    /// native endian. PngEncoder will automatically convert to big endian as required by the
+    /// underlying PNG format.
```
Comment (fintelia) [inline src/codecs/png.rs:585]: Adding this doc-comment makes rustdoc expand ImageEncoder::write_image by default.

File: src/codecs/png.rs [hunk 2]
```diff
@@ -532,6 +532,9 @@ impl<W: Write> PngEncoder<W> {
     }
 
     /// Encodes the image `data` that has dimensions `width` and `height` and `ColorType` `c`.
+    ///
+    /// Expects data in big endian.
+    #[deprecated = "Use `PngEncoder::write_image` instead. Beware that `write_image` has a different endianness convention"]
     pub fn encode(self, data: &[u8], width: u32, height: u32, color: ColorType) -> ImageResult<()> {
```
Comment (HeroicKatora) [inline src/codecs/png.rs:538]: Maybe add `#[doc(hidden)]`?

Comment (fintelia) [reply src/codecs/png.rs:538]: Hm, if we do that then users won't be able to see that the function expected big endian. (Older releases didn't mention it)

Your call

Comment (HeroicKatora) [reply src/codecs/png.rs:512]: I guess we can have it deprecated for a couple of versions before hiding it :+1:

**Review by HeroicKatora — APPROVED** [review]

**Pull request merged**